pilot/pkg/bootstrap/server.go

```diff
@@ -1281,3 +1288,73 @@ func (s *Server) initStatusManager(_ *PilotArgs) {
 		return nil
 	})
 }
+
+// initLocalSecretListener creates a new secret listener to listen for
+// secrets which represent new multi cluster participating clusters in a mesh.
+func (s *Server) initLocalSecretListener(args *PilotArgs) {
+	// Check if this is an instance of Istiod running as external istiod
+	// The logic is based on the following:
+	// 1. command istioctl secret create will always create a secret named
+	//    istio-kubeconfig when secret type is config, to indicate that a cluster
+	//    gets used as a config cluster, the secret type must be config
+	// 2. this secret is required to be added to the namespace where istiod will be
+	//    running, otherwise, istiod wont run as an external istiod
+	// 3. this secret has to be mounted to a location to which the env KUBECONFIG points
+	// 4. due to the structure of the secret, this secret will be always mounted
+	//    as a file named config such as /var/run/secrets/remote/config
+	// 5. this kubeconfig is different from loading credentials off of a service account
+	//    which is how non-external istiod gets configured normally.
+	// Based on the above, this method will just check if the file which the value of
+	// KUBECONFIG points exists and if features.ExternalIstiod is true. With these two
+	// conditions met we should be confident that this is an external Istiod instance.
+
+	// Inspect if the KUBECONFIG pointed location has a file
+	kubeconfigVar := env.StringVar{Var: env.Var{
+		Name: "KUBECONFIG", DefaultValue: "",
+		Description: "", Type: env.STRING,
+	}}
+	if !file.Exists(kubeconfigVar.Get()) {
```

Comment:
What does istiod do during startup to decide if it is using an external config cluster or local cluster for config? Same check as this?

Reply:
Not sure i understand here, to istiod config cluster is a normal cluster. How could `KUBECONFIG` env related

Reply:
Istiod doesn't really know the difference. Mainly just the EXTERNAL_ISTIOD feature flag

Reply:
@stevenctl in that case, how do you suggest istiod can decide whether or not it needs to listen to a different cluster for the remote secrets?

Reply:
A new feature flag that gives the "local" kubeconfig, only to be used for this secret watching purpose? 

If _all_ clusters (including the config cluster) have to be referenced by secret, I think we will have to overhaul our bootstrap code.

Reply:
I think a lot of the changes made in `bootstrap` should really be handlers for `secretcontroller`.

Reply:
My opinion is not tio use `KUBECONFIG` to indicate `local` here